packages/react/src/core/CodeMirrorEditor/ImageContent.tsx

```diff
@@ -0,0 +1,24 @@
+/**
+ * Based on https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types
+ * Removing svg, which is not binary.
+ */
+const binaryImageExtensions = new Set(['apng', 'png', 'avif', 'gif', 'jpg', 'jpeg', 'jfif', 'pjpeg', 'pjp', 'webp']);
+
+export function uint8ArrayToDataUrl(arr: Uint8Array) {
+  return URL.createObjectURL(new Blob([arr.buffer]));
+}
+export function isBinaryImagePath(filePath: string) {
+  const split = filePath.split('.');
+
+  if (split.length < 2) {
+    return false;
+  }
+
+  const ext = split[split.length - 1];
+
+  return binaryImageExtensions.has(ext);
+}
+
+export function ImageContent({ src }: { src: string }) {
+  return <img src={src} />;
+}
```

Comment:
> Problems left:
> somehow the editor displays code from another file alongside the image

The existing `<BinaryContent />` does too, but it uses `position: absolute; inset: 0` to cover the text area. We could do same here:

```suggestion
export function ImageContent({ src }: { src: string }) {
  return (
    <div className="flex items-center justify-center absolute inset-0 z-10 bg-tk-elements-app-backgroundColor">
      <img src={src} />
    </div>
  );
}
```

This isn't actually good approach as the text content of previous file is still present in DOM. It's part of tab and read order so users can actually navigate to this invisible element. But as this bug is already present in TutorialKit we can ignore it on this PR. 🤷‍♂️

Reply:
I think this is a leftover of a time where we were trying to cache the `EditorView` from CodeMirror which I don't think is necessary. We should change the logic so that the `CodeMirrorEditor` is not rendered in that case.